components/event-sources/upgrade-job/process/clean_up.go

```diff
@@ -0,0 +1,32 @@
+package process
+
+import (
+	"github.com/pkg/errors"
+)
+
+var _ Step = &CleanUpStep{}
+
+type CleanUpStep struct {
+	name    string
+	process *Process
+}
+
+func NewCleanUp(p *Process) CleanUpStep {
+	return CleanUpStep{
+		name:    "Cleanup of the backed up Config Map",
+		process: p,
+	}
+}
+
+func (s CleanUpStep) Do() error {
+	err := s.process.Clients.ConfigMap.Delete(BackedUpConfigMapNamespace, BackedUpConfigMapName)
+	if err != nil {
```

Comment:
In case of error, does it make sense to al least label the ConfigMap with something indicates that migration actually was done? Because only the cleanup in the end failed, but actually the migration was done. This is purely for troubleshooting.

Reply:
In such a case, the job fails anyway hence a human intervention is needed to fix it. The label won't add any value IMO.

Reply:
The label will help whoever troubleshooting what happens that actual migration is done, but only cleaning up the ConfigMap failed.